lib/pengine/tests/native/native_find_rsc_test.c

```diff
@@ -0,0 +1,350 @@
+
+    /* Fails because pe_find_current is required if a node is given. */
+    assert_null(native_find_rsc(public_ip, "Public-IP", cluster02, 0));
+
+    /* Passes because Public-IP is running on cluster02. */
+    assert_ptr_equal(public_ip, native_find_rsc(public_ip, "Public-IP", cluster02, pe_find_current));
+}
+
+static void
+inactive_group_member_rsc(void **state) {
+    pe_resource_t *inactive_dummy_1 = NULL;
+
+    /* Find the "inactive-dummy-1" resource, a member of "inactive-group". */
+    for (GList *iter = data_set->resources; iter != NULL && inactive_dummy_1 == NULL; iter = iter->next) {
+        pe_resource_t *rsc = (pe_resource_t *) iter->data;
+
+        if (strcmp(rsc->id, "inactive-group") == 0) {
+            for (GList *child = rsc->children; child != NULL; child = child->next) {
+                rsc = (pe_resource_t *) child->data;
+
+                if (strcmp(rsc->id, "inactive-dummy-1") == 0) {
+                    inactive_dummy_1 = rsc;
+                    break;
+                }
+            }
+        }
+    }
+
+    assert_non_null(inactive_dummy_1);
+
+    /* Passes because NULL was passed for node, regardless of flags. */
+    assert_ptr_equal(inactive_dummy_1, native_find_rsc(inactive_dummy_1, "inactive-dummy-1", NULL, 0));
+    assert_ptr_equal(inactive_dummy_1, native_find_rsc(inactive_dummy_1, "inactive-dummy-1", NULL, pe_find_current));
+
+    /* Fails because resource is not a clone. */
+    assert_null(native_find_rsc(inactive_dummy_1, "inactive-dummy-1", NULL, pe_find_clone));
+    assert_null(native_find_rsc(inactive_dummy_1, "inactive-dummy-1", cluster01, pe_find_clone));
+
+    /* Fails because inactive-dummy-1 is not running. */
+    assert_null(native_find_rsc(inactive_dummy_1, "inactive-dummy-1", cluster01, pe_find_current));
+    assert_null(native_find_rsc(inactive_dummy_1, "inactive-dummy-1", cluster02, pe_find_current));
+
+    /* Passes because of flags. */
+    assert_ptr_equal(inactive_dummy_1, native_find_rsc(inactive_dummy_1, "inactive-dummy-1", cluster01, pe_find_inactive));
+    /* Passes because of flags. */
+    assert_ptr_equal(inactive_dummy_1, native_find_rsc(inactive_dummy_1, "inactive-dummy-1", cluster02, pe_find_inactive));
+}
+
+static void
+clone_rsc(void **state) {
+    pe_resource_t *ping_clone = NULL;
+
+    /* Find the "ping-clone" resource. */
+    for (GList *iter = data_set->resources; iter != NULL; iter = iter->next) {
+        pe_resource_t *rsc = (pe_resource_t *) iter->data;
+
+        if (strcmp(rsc->id, "ping-clone") == 0) {
+            ping_clone = rsc;
+            break;
+        }
+    }
+
+    assert_non_null(ping_clone);
+
+    /* Passes because NULL was passed for node, regardless of flags. */
+    assert_ptr_equal(ping_clone, native_find_rsc(ping_clone, "ping-clone", NULL, 0));
+    assert_ptr_equal(ping_clone, native_find_rsc(ping_clone, "ping-clone", NULL, pe_find_current));
```

Comment:
pe_find_clone with NULL node is worthwhile too

Reply:
Added.